src/plugins/discover/public/application/main/components/sidebar/discover_sidebar_responsive.tsx

```diff
@@ -171,6 +177,53 @@ export function DiscoverSidebarResponsive(props: DiscoverSidebarResponsiveProps)
   }, []);
 
   const { dataViewFieldEditor } = props.services;
+  const { availableFields$ } = props;
+
+  useEffect(
+    () => {
+      // For an external embeddable like the Field stats
+      // it is useful to know what fields are populated in the docs fetched
+      // or what fields are selected by the user
+
+      const documents = documentState.result;
+      const availableFields: DataViewField[] = [];
+      const fieldCnts = fieldCounts.current;
+      if (!fieldCnts) return;
+
+      const newFields = getIndexPatternFieldList(selectedIndexPattern, fieldCnts);
+
+      // If fields are selected, we only need to broadcast the selected fields
+      if (props.columns.length > 0) {
+        newFields.forEach((field) => {
+          if (props.columns.findIndex((c) => c === field.name) > -1) {
+            availableFields.push(field);
+          }
+        });
+      } else {
+        // Else, go through the docs and find out what fields are populated/available
+        if (documents && fieldCounts?.current !== null) {
+          newFields.forEach((field) => {
+            if (fieldCnts[field.name] > 0) {
+              availableFields.push(field);
+            }
+          });
+        }
+      }
+
+      availableFields$.next({
+        fetchStatus: FetchStatus.COMPLETE,
+        fields: availableFields,
+      });
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [
+      selectedIndexPattern,
+      availableFields$,
+      fieldCounts.current,
+      documentState.result,
+      props.columns.length,
+    ]
```

Comment:
think you should be able to remove 
```
    // eslint-disable-next-line react-hooks/exhaustive-deps
```

By using `[selectedIndexPattern, availableFields$, documentState.result, props.columns]`

Reply:
The `columns.length` here is a small performance optimization since props.columns is an array, so its reference will change every render. I didn't think it's worth triggering the effect if the user didn't add or remove the column (or if data view changes). I'm adding an extra comment here for now https://github.com/elastic/kibana/pull/123003/commits/bb2b61dd3f9fb3e1b3e9970f6000a17f48b872fb but I'm happy to remove this `// eslint-disable-next-line` completely and replace with props.columns instead.

Reply:
fine with that solution, thx for explaining 👍